arangod/Aql/IResearchViewOptimizerRules.cpp

```diff
@@ -294,35 +308,47 @@ bool optimizeScoreSort(IResearchViewNode& viewNode, ExecutionPlan* plan) {
   std::vector<std::pair<size_t, bool>> scoresSort;
   for (auto const& sort : sortElements) {
     TRI_ASSERT(sort.var);
-    auto varSetBy = plan->getVarSetBy(sort.var->id);
+
+    auto const* varSetBy = plan->getVarSetBy(sort.var->id);
     TRI_ASSERT(varSetBy);
-    arangodb::aql::Variable const* sortVariable{};
-    if (varSetBy->getType() == ExecutionNode::CALCULATION) {
-      auto* calc = ExecutionNode::castTo<CalculationNode*>(varSetBy);
-      TRI_ASSERT(calc->expression());
-      auto astCalcNode = calc->expression()->node();
-      if (!astCalcNode ||
-          astCalcNode->type != AstNodeType::NODE_TYPE_REFERENCE) {
-        // Not a reference?  Seems that it is not
-        // something produced by during search function replacement.
-        // e.g. it is expected to be LET sortVar = scorerVar;
-        // Definately not something we could handle.
-        return false;
+
+    aql::Variable const* sortVariable{};
+    switch (varSetBy->getType()) {
+      case ExecutionNode::CALCULATION: {
+        auto* calc = ExecutionNode::castTo<CalculationNode const*>(varSetBy);
+        TRI_ASSERT(calc->expression());
+
+        auto const* astCalcNode = calc->expression()->node();
+        if (!astCalcNode ||
+            astCalcNode->type != AstNodeType::NODE_TYPE_REFERENCE) {
+          // Not a reference?  Seems that it is not
+          // something produced by during search function replacement.
+          // e.g. it is expected to be LET sortVar = scorerVar;
+          // Definately not something we could handle.
+          return false;
+        }
+
+        sortVariable =
+            reinterpret_cast<aql::Variable const*>(astCalcNode->getData());
+        TRI_ASSERT(sortVariable);
+        break;
       }
-      sortVariable = reinterpret_cast<arangodb::aql::Variable const*>(
-          astCalcNode->getData());
-      TRI_ASSERT(sortVariable);
-    } else {
-      // FIXME (Dronplane): here we should deal with stored
-      //                    values when we will support such optimization
-      return false;
+      case ExecutionNode::ENUMERATE_IRESEARCH_VIEW:
+        // FIXME (Dronplane): here we should deal with stored
+        //                    values when we will support such optimization
+        [[fallthrough]];
+      default:
+        TRI_ASSERT(false);
+        return false;
     }
 
-    auto s = std::find_if(scorers.begin(), scorers.end(),
-                          [sortVariable](SearchFunc const& t) {
-                            return t.var->id == sortVariable->id;
-                          });
-    if (s == scorers.end()) {
+    TRI_ASSERT(sortVariable);
```

Comment:
remove it?
We have         TRI_ASSERT(false); and         TRI_ASSERT(sortVariable); above